app/assets/javascripts/Components/markdown_edit.jsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import MarkdownPreview from "./markdown_preview";
+import {Tab, Tabs, TabList, TabPanel} from "react-tabs";
+
+import PropTypes from "prop-types";
+
+export default class MarkdownEdit extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      content: this.props.content,
```

Comment:
Okay so overall it's not great to duplicate props and state values. You can remove this state and have this component rely entirely on its `props.content` value, and pass `this.props.handleChange` directly to the underlying `textarea` field.